docs/configuration.md

```diff
@@ -0,0 +1,64 @@
+# Configuration
+
+Here is the list of available properties:
+
+| **Property Name**               | **Default Value** | **Explanation**                                                                                                                                                                             |
+|---------------------------------|-------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `shutdownTimeoutMs`             | 90,000ms          | Maximum number of milliseconds a resource shutting down via `BaseReactiveResourceShutdownStrategy` should take to shutdown.                                                                 |
+| `clientsReactionTimeMs`         | 30,000ms          | How many milliseconds to wait after calling `prepareForShutdown` on each `GracefulShutdownStrategy` so that all clients understand that they should not call this application node anymore. |
+| `strategiesCheckIntervalTimeMs` | 5,000ms           | How long to wait in milliseconds between checking the `canShutdown` flag of each `GracefulShutdownStrategy` being executed.                                                                 |
+| `resourceCheckIntervalTimeMs`   | 250ms             | How long to wait in milliseconds between checking termination status of a resource shutting down via the `BaseReactiveResourceShutdownStrategy`.                                            |
+
+To configure the properties, edit your `application-*.yml` file in your `resources` folder as such:
+```yaml
+tw-graceful-shutdown:
+  shutdown-timeout-ms: <INTEGER>
+  client-reaction-time-ms: <INTEGER>
+  strategies-check-interval-time-ms: <INTEGER>
+  resource-check-internal-time-ms: <INTEGER>
+```
+
+### Health Indicator
+```yaml
+tw-graceful-shutdown:
+  health-indicator:
+    enabled: [true|default:false]
+```
+If set to `true`, creates `GracefulShutdownHealthStrategy` and `GracefulShutdownHealthIndicator` beans conditionally on the presence of the `org.springframework.boot.actuate.health.AbstractHealthIndicator` class.
+
+### Kagkarlsson DB Scheduler
+```yaml
+tw-graceful-shutdown:
+  kagkarlsson-db-scheduler:
+    enabled: [true|default:false]
```

Comment:
Is default value actually - false? By default we don't have application.yml with configuration for this project. So that means this property is missing. In that case strategy beans have this annotation: `@ConditionalOnProperty(value = "tw-graceful-shutdown.kagkarlsson-db-scheduler.enabled", matchIfMissing = true)`. So if property is not configured explicitly - it is enabled. It lloks like default value is true. If this notion `true|default:false` means the same, than it is not very clear and probably could be changed to `true (default) | false`.